main.go

```diff
@@ -119,14 +138,9 @@ func (c *dnsimpleDNSProviderSolver) getDomainAndEntry(ch *v1alpha1.ChallengeRequ
 }
 
 func (c *dnsimpleDNSProviderSolver) getExistingRecord(cfg *dnsimpleDNSProviderConfig, client *dnsimple.Client, accountID string, zoneName string, entry string, key string) (*dnsimple.ZoneRecord, error) {
-	zone, err := client.Zones.GetZone(context.Background(), accountID, zoneName)
```

Comment:
Why not keep this safeguard?
It seems even more likely to me that tokens could have the wrong scope and that the configured zone is not accessible by the client.

Reply:
@moll-re Would this really be an additional safeguard? To me it doesn't look like that. If the token does not have the appropriate permission it will fail on ListRecords instead of GetZone. To me this seem fine. Further this will save one HTTP request.

The only thing I ask my self is why it was implemented like this in the first place. Why look up zone (`zone.Data.Name`) if you already have the zoneName available. Are zone.Data.Name and zoneName always the same?

Reply:
Also @IntegralProgrammer mentioned that this call does not work with user tokens:
> I looked at PR(https://github.com/puzzle/cert-manager-webhook-dnsimple/pull/29) and, although I didn't test it yet it, I anticipate that it would fail at https://github.com/dvob/cert-manager-webhook-dnsimple/blob/95329b721c2c353d4d947070cf34b8f2e0fef0f2/main.go#L141 given that I had to bypass that call in my branch https://github.com/IntegralProgrammer/cert-manager-webhook-dnsimple/blob/ca1b10acea4654bc21385e93dd0a0e240c3eb880/main.go#L122-L138 that being said, that change could easily be added to PR(https://github.com/puzzle/cert-manager-webhook-dnsimple/pull/29) as well.

Reply:
Ah, thanks for pointing this out!